sdk/include/ds/pointer.h

```diff
@@ -0,0 +1,268 @@
+// Copyright Microsoft and CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file Pointer utilities
+ */
+
+#pragma once
+
+#include <cheri.hh>
+#include <concepts>
+
+namespace ds::pointer
+{
+
+	/**
+	 * Offset a pointer by a number of bytes.  The return type must be
+	 * explicitly specified by the caller.  The type of the displacement offset
+	 * (Offset) is templated so that we can accept both signed and unsigned
+	 * offsets.
+	 */
+	template<typename T, typename Offset = size_t, typename U>
+	static inline __always_inline T *offset(U *base, Offset offset)
```

Comment:
```suggestion
	template<typename T, typename U>
	static inline __always_inline T *offset(U *base, std::integral auto offset)
```

Not sure if we have the `std::integral` concept, if not then it can be dropped.